src/validators/property-order.js

```diff
@@ -78,7 +78,11 @@ const isInPreferredOrder = function(packageJsonData, userPreferredNodeOrder) {
 
     const preferredOrderPosition = filteredPreferredOrderMap.get(currentPkgJsonProperty);
 
-    if (preferredOrderPosition !== keyIndex) {
+    if (preferredOrderPosition > keyIndex) {
+      isValid = false;
+      msg = `Please move "${fltrdPreferredNodeOrder[preferredOrderPosition - one]}" before "${currentPkgJsonProperty}".`;
+      break;
+    } else if (preferredOrderPosition < keyIndex) {
```

Comment:
Hey @moshest is it possible to get in this branch?

Reply:
I didn't understand what you ask..

Reply:
My apologizes! I am curious if it is possible to have `preferredOrderPosition` be less than `keyIndex`?

We are always iterating over the filtered keys in the package.json file and comparing them to the matching properties in the preferred property order list. For example:

|keyIndex|package.json key|
|-|-|
|0|name|
|1|version|
|2|description|
|3|devDependencies|
|4|dependencies|

|keyIndex|package.json key|
|-|-|
|0|devDependencies|
|1|dependencies|
|2|version|
|3|description|
|4|name|

In this scenario, the time through the loop `keyIndex` will be `0`. `0` is `name` in package.json. When `name` is looked up in the preferred list we get a value of `4`. `preferredOrderPosition` is greater than `keyIndex`. This will throw an error saying that `Please move "description" before "name"`  and the user will correct it. 

Now the file will look like

|keyIndex|package.json key|
|-|-|
|0|description|
|1|name|
|2|version|
|3|devDependencies|
|4|dependencies|

|keyIndex|package.json key|
|-|-|
|0|devDependencies|
|1|dependencies|
|2|version|
|3|description|
|4|name|

The second time through the loop `keyIndex` will be `0`. `0` is `description` in package.json. When `description` is looked up in the preferred list we get a value of `3`. `preferredOrderPosition` is greater than `keyIndex`. This will throw an error saying that `Please move "version" before "description"`  and the user will correct it. 

Now the file will look like

|keyIndex|package.json key|
|-|-|
|0|version|
|1|description|
|2|name|
|3|devDependencies|
|4|dependencies|

|keyIndex|package.json key|
|-|-|
|0|devDependencies|
|1|dependencies|
|2|version|
|3|description|
|4|name|

The third time through the loop `keyIndex` will be `0`. `0` is `version` in package.json. When `version ` is looked up in the preferred list we get a value of `2`. `preferredOrderPosition` is greater than `keyIndex`. This will throw an error saying that `Please move "dependencies" before "version"`  and the user will correct it. 

Now the file will look like

|keyIndex|package.json key|
|-|-|
|0|dependencies|
|1|version|
|2|description|
|3|name|
|4|devDependencies|

|keyIndex|package.json key|
|-|-|
|0|devDependencies|
|1|dependencies|
|2|version|
|3|description|
|4|name|

The third time through the loop `keyIndex` will be `0`. `0` is `dependencies` in package.json. When `dependencies` is looked up in the preferred list we get a value of `1`. `preferredOrderPosition` is greater than `keyIndex`. This will throw an error saying that `Please move "devDependencies" before "dependencies"`  and the user will correct it. 

The next run everything will be in order, so no error will be thrown. 

I'm not sure if it possible to have `preferredOrderPosition` be less than `keyIndex` since we are iterating over package.json keys and we are filtering each list to remove properties that aren't in both lists. Sorry for the long post 🤓

Reply:
Yes, I think you right.

Reply:
I like the idea of changing the message to before instead of after. What do you think about updating the tests and merging this PR in?

Reply:
Hey @moshest are you still supportive of moving this message before instead of after?

Reply:
Hi, my experience when sorting the properties has a bit frustrating, I remember that every time I had to move up one property but then I had to move other property above it so it's like never ending sorting.

Take for example the first example, the exception said `Please move "name" before "description"`. So the user put it right before `description`, but `description` is not in the right place either, so all the process start again.

A better approach will be to ask for the user to `Please move "name" before "devDependencies"`. Then the `name` is in the right place and the user can continue sorting the other properties.

I'm sorry for not responding regarding the PR until now. I'm not feeling comfortable diving into the code yet.

Reply:
No worries at all. I really appreciate your great feedback and willingness to help make this project better. I can take a shot at updating the logic to before instead of after. Are you okay if I decline this PR after I’m done?

Reply:
Yes, sure.